packages/transaction-controller/src/types.ts

```diff
@@ -398,6 +398,17 @@ type TransactionMetaBase = {
    */
   txParams: TransactionParams;
 
+  approvalTxParams?: {
```

Comment:
I worry this is misleading since the transaction params are what is defined in the Ethereum standard, where as this is ultimately just metadata used for swap approvals.

Is all this not already available in this type via `txParams`, `type` and `swapMetaData`?